src/main/java/graphql/schema/idl/SchemaGeneratorDirectiveHelper.java

```diff
@@ -133,16 +134,19 @@ private GraphqlElementParentTree buildRuntimeTree(GraphQLSchemaElement... elemen
         }).collect(toList());
     }
 
-    private List<GraphQLFieldDefinition> wireFields(GraphQLFieldsContainer fieldsContainer, NamedNode fieldsContainerNode, Parameters params) {
+    private List<GraphQLFieldDefinition> wireFields(GraphQLFieldsContainer fieldsContainer, NamedNode<?> fieldsContainerNode, Parameters params) {
         return fieldsContainer.getFieldDefinitions().stream().map(fieldDefinition -> {
 
             // and for each argument in the fieldDefinition run the wiring for them - and note that they can change
+            List<GraphQLArgument> startingArgs = fieldDefinition.getArguments();
             List<GraphQLArgument> newArgs = wireArguments(fieldDefinition, fieldsContainer, fieldsContainerNode, params, fieldDefinition);
 
-            // they may have changed the arguments to the fieldDefinition so reflect that
-            fieldDefinition = fieldDefinition.transform(builder -> builder.clearArguments().arguments(newArgs));
+            if (isNotTheSameObjects(startingArgs, newArgs)) {
```

Comment:
I changed this so that IF the node did not change, we just return the old node - slight optimisation in tree building